File: workflow_shell/git/cherry-pick.py
from ..util import run


command_string = "gcp"
command_help = "Takes one argument, the commit SHA. Cherry picks that commit"
short_help = "Git - cherry-picks a SHA"
command_arguments = ["sha"]
argument_required = True


def main(*args, **kwargs):
    argument = kwargs.get(command_arguments[0])
    run(["git", "cherry-pick", argument], True)

File: src/commands/git/CherryPick.ts
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';
import { Names } from '../../common/interface/Names';
import { Option } from '../../common/interface/Option';

const options: Option[] = [];
const argumentss: Argument[] = [Argument.COMMIT_SHA];

export class Command extends SuperCommand {
  description: string = 'Takes one argument, the commit SHA. Cherry picks that commit.';
  alias: string = 'gcp';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      await executeCommand('git', ['cherry-pick', this.input[Names.COMMIT_SHA]]);
    });
  }
}
